gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposeOffchainConfig.ts

```diff
@@ -107,12 +118,33 @@ export default class ProposeOffchainConfig extends SolanaCommand {
     return input
   }
 
-  makeInput = (userInput: any): Input => {
+  makeInput = async (userInput: any): Promise<Input> => {
     if (userInput) return userInput as Input
     const rdd = RDD.load(this.flags.network, this.flags.rdd)
+
+    const ocr2 = getContract(CONTRACT_LIST.OCR_2, '')
+    const address = ocr2.programId.toString()
+    this.program = this.loadProgram(ocr2.idl, address)
+
+    const userSecret = this.flags.secret
+    if (!!userSecret) {
+      logger.info(`Using given random secret: ${userSecret}`)
+    }
+
+    const offchainConfig = ProposeOffchainConfig.makeInputFromRDD(rdd, this.args[0])
+    // process.env.SECRET is required on the command
+    const { offchainConfig: serializedOffchainConfig, randomSecret } = await serializeOffchainConfig(
+      offchainConfig,
+      process.env.SECRET!,
+      userSecret,
+    )
+    this.randomSecret = randomSecret
+
     return {
-      offchainConfig: ProposeOffchainConfig.makeInputFromRDD(rdd, this.args[0]),
+      offchainConfig,
+      serializedOffchainConfig,
```

Comment:
We can't ask the user to build the `serializedOffchainConfig`. This cannot be part of the `Input`.
An idea could be having `makeContractInput`, similar to what we have on Terra. The serialization would happen there.